packages/sanity/src/core/studio/upsell/upsellDescriptionSerializer/helpers.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  isPortableTextTextBlock,
+  PortableTextObject,
+  type PortableTextBlock,
+  PortableTextSpan,
+} from '@sanity/types'
+
+const TRANSFORMATIONS: {
+  [type: string]: (
+    child: PortableTextObject | PortableTextSpan,
+    idx: number,
+    block: PortableTextBlock,
+  ) => PortableTextObject
+} = {
+  inlineIcon: (child, idx, block) => {
+    if (!isPortableTextTextBlock(block)) return block
+    const hasTextLeft = Boolean(
```

Comment:
Do these array accesses need to be checked/guarded in some way? Just wondering if accessing [-1] will ever happen/be a problem or if accessing [length+1] will ever throw an error

Reply:
The upper check `isPortableTextTextBlock` is checking for children to be defined and to be of type array. 
And later when accessing the children `block.children[idx + 1]?._type` will work as a safeguard in case it's trying to access to an undefined children